### PR TITLE
Add to UserGuide known issues and specifications, DeveloperGuide planned enhancements

### PR DESCRIPTION
**Pull request by realqijun**
# Resolves 
* #217,
* #219,
* #220,
* #225,
* #261

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -900,3 +901,9 @@ The inverse of adding to favourites. The "star" in the top-right corner should c
       ```
 
 [back to top](#internbuddy-developer-guide)
+
+## **Appendix: Planned Enhancements**
+
+Team size: 5
+
+1. Expanding the set of special characters allowed in the `NAME` field to support a wider range of company names and applications. Currently, only alphanumeric characters are permitted, which means names like "Ernst & Young" and "UI/UX developer" cannot be accurately represented.
```
Comment (blanklogic) [inline docs/DeveloperGuide.md:909]: Suggestion: 
```md
**Make `NAME` field less restrictive**: Currently, only alphanumeric characters are permitted, which means names like "Ernst & Young" and "UI/UX developer" cannot be accurately represented. We plan to expand the set of special characters allowed in the `NAME` field to support a wider range of company names and applications. 
```

**Review by blanklogic — APPROVED** [review]
Minor nits for Planned Enhancements but other than that LGTM!

**Comment by uniqly** [comment]
This PR also addresses #228

**Review by uniqly — APPROVED** [review]
LGTM!

**Pull request merged**